### PR TITLE
Refactor `ResNetBackbone` for readability

### LINKED ISSUE #1808: Refactor `ResNetBackbone` for readability
Our backbone modeling files are very large and hard to read. @LukeWood proposed two changes in #1745:

1. move aliases to their own file
2. move helpers below model declarations

#1745 provides a nice template for this change for `MobileNetBackbone`. Replicate this change for `keras_cv/models/backbones/resnet_v1/`.

### PR DESCRIPTION
**Pull request by ID6109**
# What does this PR do?

Fixes #1808

### CODE REVIEW

File: keras_cv/models/backbones/resnet_v1/resnet_v1_backbone.py [hunk 1]
```diff
@@ -219,333 +355,3 @@ def apply_stack(
...
-    @classproperty
-    def presets_with_weights(cls):
-        """Dictionary of preset names and configurations that include
-        weights."""
-        return {}
-
-
-class ResNet152Backbone(ResNetBackbone):
-    def __new__(
-        self,
```
Comment (ID6109) [inline keras_cv/models/backbones/resnet_v1/resnet_v1_backbone.py:519]: I'm presuming this was an error in our original implementation.

**Review by ianstenbit — APPROVED** [review]
Thanks for the PR -- nice that you caught a few docstring fixes in here too!

**Comment by ianstenbit** [comment]
Test failures are unrelated to this PR. See #1855.

Once that is merged, we can merge from master into this dev branch and then CI should be happy.

**Pull request merged**